cloud/google/metadata.go

```diff
@@ -119,8 +111,7 @@ NODE_TAG="$CLUSTER_NAME-worker"
 const nodeEnvironmentVars = `
 #!/bin/bash
 KUBELET_VERSION={{ .Machine.Spec.Versions.Kubelet }}
-TOKEN={{ .Token }}
-MASTER={{ .MasterEndpoint }}
```

Comment:
I am not sure of the direction we are going if there was a doc I could read that would make it more clear that would he helpful -- I don't think it is hard to add this back if needed.

Reply:
@jessicaochen I agree. We want to enable the use of kubeadm, but we don't want to be tied to it. What is in the join command that is not encapsulated by the token and master variables? Perhaps that is what should be added.

Reply:
Hello @jessicaochen, @krousey 

The new bit is the parameter "--discovery-token-ca-cert-hash" which "allows for bootstrapping nodes to securely discover a root of trust for the master". You can read more about it at the link below at [1]. By passing that value in we are able to remove the parameter "--discovery-token-unsafe-skip-ca-verification" from this command.

It's not required that we use the join command returned by kubeadm I felt like it was a bit better to just use what kubeadm recommends as we end up basically stitching that together. The more variables that are passed into the script the more there is to get wrong, etc, plus I thought it might magically do the right thing in future scenarios such as multi-master / HA.

I'll update the PR to retain the passing in and use of the Master IP address. We can parse the hash verification out of the join command and retain passing the Master IP into the join command.

[1] https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-join/